src/api/repos/user-repo.ts

```diff
@@ -66,6 +66,25 @@ const getUserFavoriteActivityList = async (id: string) => {
     };
 };
 
+const checkActivityFavoritedByUser = async (id: string, activity: string) => {
+    await User.sync();  
+    await Activity.sync();
+    const user = (await getUserById(id)).user;
+    if (!user) {
+        throw new APIError(`Cannot find User with firebase_id ${id}`, 'checkActivityFavoritedByUser', httpStatusCode.CONFLICT);
+    }
+
+    try{
+        const activityId = parseInt(activity)
+        return {
+            isActivityFound: user.favoriteActivities.includes(activityId) 
+        };
+    }
+    catch(err){
+        throw new APIError('Error during Activity string to number conversion', 'checkActivityFavoritedByUser', httpStatusCode.CONFLICT);
```

Comment:
Use `activityId` that was exported from `activity-schema` to validate the `activityId` before even getting to this point. so you don't have to throw this APIError